agent/src/capability/servicemanifestloader.h

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (C) 2016-2017 Pelagicore AB
+ *
+ * Permission to use, copy, modify, and/or distribute this software for
+ * any purpose with or without fee is hereby granted, provided that the
+ * above copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL
+ * WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR
+ * BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES
+ * OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS,
+ * WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION,
+ * ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS
+ * SOFTWARE.
+ *
+ * For further information see LICENSE
+ */
+
+
+#pragma once
+
+#include "softwarecontainer-common.h"
+#include "configstoreerror.h"
+
+#include <jsonparser.h>
+
+namespace softwarecontainer {
+
+/*
+ * Abstract interface to a service manifest loader.
+ *
+ * The loader will either be a production loader that loads a json file
+ * from a file on the system, or a test loader that loads json from a string.
+ */
+class ServiceManifestLoader
+{
+public:
+    // Enforce inheriting classes to initialize service manifest source member
+    ServiceManifestLoader() = delete;
+    ServiceManifestLoader(const std::string &source) : m_source(source) {}
+
+    virtual ~ServiceManifestLoader() {
+        for (json_t *item : m_content) {
+            json_decref(item);
+        }
+    }
+
+    virtual std::vector<json_t *> loadContent() = 0; // TODO what to call this
```

Comment:
About the TODO - you don't need to call it anything, this is a pure virtual and we should expect the reader to know that since it's a common thing.

Also, can't this method throw exceptions? If so, we should state that  somewhere so the user of this interface knows what to expect.